[DOCS][101] Create Elasticsearch basics section, refactor quickstarts section

leemthompo opened the pull request:
##  ⚠️ Use the [markup preview](https://app.markup.io/markup/41a10056-a6b7-4c0b-a5bd-1caeda4276d0?view_mode=Desktop) for general feedback

Follow up to https://github.com/elastic/elasticsearch/pull/112213 and https://github.com/elastic/elasticsearch/pull/112394 

###  👀 [Standard PR URL preview](https://elasticsearch_bk_112436.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/elasticsearch-intro.html)

☝️ See the live page preview 


## Summary of changes

ℹ️ `docs/reference/intro.asciidoc` has big diff, need to expand under **Files changed** to see changes

- Move "What is Elasticsearch?" content under bigger **Elasticsearch basics** section at top of docs nav.
  - Split out deployment options under **Run Elasticsearch** on its own page
  - Overhaul **Search and analyze** page
  - Add new page: **Add data to Elasticsearch**
  - ~Add new page: **Start with the REST API**~
  - Rename Scalability and resilience page to "**Plan for production**" (content will be refactored later)
- Refactor quickstart landing page to accommodate future quickstarts
  - Move local dev instructions away now that we have better signposting in deployment options section
    - ℹ️ This will be replaced by `start-local` soon

> [!NOTE] 
> See commit messages for more details.


## Context

- I've tried to work on all these pages at the same time because it was more efficient. However, this is now a big PR. The goal is not to merge all these changes at once. 

- Hopefully we get a URL preview and we'll use that as the basis for further discussions and can tactically update pages. 

> [!TIP]
> Definitely don't want wording edits immediately until we've agreed on how to proceed. Will try to have conversation in markup.io to enable free form commenting :)

leemthompo committed:
Create 101 section

- **index.asciidoc**:
  - Added a reference to a new document: `scalability.asciidoc`.

- **intro.asciidoc**:
  - Changed the section title from "What is {es}?" to "{es} 101".
  - Added a detailed explanation for beginners under "{es} 101", covering:
    - Basic information about Elasticsearch.
    - Key features and deployment.
    - Data storage and ingestion.
    - Data searching and analyzing.
    - Planning for scalability and resilience.
  - Updated the "Deployment options" section, adding quick start options and enhancing existing content for clarity.

leemthompo committed:
- **intro.asciidoc**:
  - Title changed from "{es} 101" to "Getting started with Elasticsearch"
  - Updated content to be more welcoming for new users, with clear guidance on how to use the docs effectively.
  - Added guidance on data ingestion, searching, analyzing data.

- **getting-started.asciidoc** (within quickstart directory):
  - Updated the guide title to "Quick start: First steps with {es}" for clarity and focus.
  - Streamlined content to specifically teach how to add a small dataset to Elasticsearch and perform basic searches, targeting new users or those in initial learning phases.
 - Removed tab widget and delegate all deployment info to deployment doc
  -

- **index.asciidoc** (within quickstart directory):
  - Renamed from "Quickstart" to "Quickstarts" .
  - Added a new "Requirements" section detailing prerequisites for using the quickstarts effectively.
  - Introduced a new "Working in Python" section pointing to Python stuff

- **run-elasticsearch-locally.asciidoc**:
  - Moved into overall Set up Elasticsearch section, now that we have nice signposting

leemthompo committed:
Merge main

leemthompo commented:
run docs-build

leemthompo committed:
Add link

leemthompo commented:
run docs-build

leemthompo committed:
Tweak ql table, mini nav shuffle

leemthompo committed:
Fix typos, links

leemthompo committed:
Move what's new page

Eventually I want this to live further down the nav
but it requires messing with heading levels more
than I can justify right now.

leemthompo committed:
Move some nav items

leemthompo committed:
Add couple links

leemthompo committed:
Update landing page getting started link

leemthompo committed:
Fix link

leemthompo committed:
Apply suggestions from feedback

leemthompo commented on docs/reference/intro.asciidoc:
```diff
@@ -1,5 +1,27 @@
 [[elasticsearch-intro]]
-== What is {es}?
+== Getting started with Elasticsearch
+++++
+<titleabbrev>Getting started</titleabbrev>
+++++
```
```suggestion
== {es} basics
```

leemthompo committed:
Rename to Elasticsearch basics

leemthompo committed:
Refactorings

leemthompo commented on docs/reference/intro.asciidoc:
```diff
@@ -155,81 +192,171 @@ You can use a combination of dynamic and explicit mapping on the same index.
+[[es-ingestion-overview]]
+=== Ingestion options
+
+There are multiple ways to ingest data into {es}.
```
Done

leemthompo commented on docs/reference/intro.asciidoc:
```diff
@@ -155,81 +192,171 @@ You can use a combination of dynamic and explicit mapping on the same index.
-insight into key metrics, patterns, and trends. Instead of just finding the
-proverbial “needle in a haystack”, aggregations enable you to answer questions
-like:
+{es} enables a host of use cases based on aggregations and analytics.
```
Done

leemthompo committed:
Merge main

leemthompo committed:
Little fixes

leemthompo committed:
The term -we you- is not a thing

leemthompo committed:
Use attributes

leemthompo committed:
Adds links to API bullet

shainaraskas commented on docs/reference/intro.asciidoc:
```diff
@@ -155,81 +184,146 @@ You can use a combination of dynamic and explicit mapping on the same index.
+[[es-ingestion-overview]]
+=== Add data to {es}
+
+There are multiple ways to ingest data into {es}, but it depends on whether you're working with time-series data or general content that doesn't have a timestamp.
```
the other doc says "timestamped data" - time series and timestamped are both very loaded terms, neither perfect. I'd probably just follow the doc we already have.

shainaraskas commented on docs/reference/intro.asciidoc:
```diff
@@ -155,81 +184,146 @@ You can use a combination of dynamic and explicit mapping on the same index.
+
+[discrete]
+[[es-ingestion-overview-time-series]]
+==== Time-series data
```
remember everyone coming to this doc is brand new - let's do a quick definition for both timestamped / general data here with a couple examples

shainaraskas commented on docs/reference/intro.asciidoc:
```diff
@@ -155,81 +184,146 @@ You can use a combination of dynamic and explicit mapping on the same index.
 
+// New html page
+[[es-ingestion-overview]]
+=== Add data to {es}
```
I'd like to see this topic get some more content from its big brother in the cloud docs. we also don't talk about pipelines, data manip, etc. which are foundational concepts

shainaraskas replied:
I don't know the ingest story super well, but I can see understanding that this is the point at which you can enrich data to customize it to your use case is really a production-side concern (esp for search use case). I view prod concerns as scaling, security, UAC ...

this is doing a lot of heavy lifting but I do think exposing those things here aligns with the 101 ask

shainaraskas commented on docs/reference/intro.asciidoc:
```diff
@@ -155,81 +184,146 @@ You can use a combination of dynamic and explicit mapping on the same index.
+
+[TIP]
+====
+If you're interested in data ingestion pipelines for time-series data, use the decision tree in the {cloud}/ec-cloud-ingest-data.html#ec-data-ingest-pipeline[Elastic Cloud docs] to understand your options.
```
```suggestion
If you're interested in data ingestion pipelines for time series data, use the decision tree in the {cloud}/ec-cloud-ingest-data.html#ec-data-ingest-pipeline[Elastic Cloud docs] to understand your options.
```

shainaraskas replied:
a general note ... I'd like to see one of these become the source of truth and the other disappear. this feels like another case of making more worse versions of content that will mess with SEO

shainaraskas replied:
coming back to this: I think that the ask for detail is mostly to solve for our bad IA. in future we will have a dedicated ingest section that can do heavier lifting, but I think detail in this area is what people can use

shainaraskas commented on docs/reference/intro.asciidoc:
```diff
@@ -155,81 +184,146 @@ You can use a combination of dynamic and explicit mapping on the same index.
++
+[TIP]
+====
+Once you get past the very basics, you'll want to use the <<docs-bulk,Bulk API>> for efficiency.
```
```suggestion
After you get past the basics, you should use the <<docs-bulk,Bulk API>> for efficiency.
```

shainaraskas commented on docs/reference/intro.asciidoc:
```diff
@@ -155,81 +184,146 @@ You can use a combination of dynamic and explicit mapping on the same index.
+====
+* {kibana-ref}/connect-to-elasticsearch.html#upload-data-kibana[File upload]: Use the {kib} File Uploader to upload and index CSV, JSON, and log files.
+* {kibana-ref}/connect-to-elasticsearch.html#_add_sample_data[Sample data]: Load sample data sets into your {es} cluster using {kib}.
+* {enterprise-search-ref}/crawler.html[Web Crawler]: Extract and index web page content into {es} documents.
```
```suggestion
* {enterprise-search-ref}/crawler.html[Web crawler]: Extract and index web page content into {es} documents.
```

shainaraskas commented on docs/reference/intro.asciidoc:
```diff
@@ -155,81 +184,146 @@ You can use a combination of dynamic and explicit mapping on the same index.
+====
+Once you get past the very basics, you'll want to use the <<docs-bulk,Bulk API>> for efficiency.
+====
+* {kibana-ref}/connect-to-elasticsearch.html#upload-data-kibana[File upload]: Use the {kib} File Uploader to upload and index CSV, JSON, and log files.
```
```suggestion
* {kibana-ref}/connect-to-elasticsearch.html#upload-data-kibana[File upload]: Use the {kib} file uploader to upload and index CSV, JSON, and log files.
```

shainaraskas commented on docs/reference/intro.asciidoc:
```diff
@@ -155,81 +184,146 @@ You can use a combination of dynamic and explicit mapping on the same index.
-While you can use {es} as a document store and retrieve documents and their
-metadata, the real power comes from being able to easily access the full suite
-of search capabilities built on the Apache Lucene search engine library.
+You can use {es} as a basic document store to simply retrieve documents and their
```
preference for no "simply" ([reasoning](https://blog.knowledgeowl.com/blog/posts/dont-say-simply-jim-fisher/))

shainaraskas commented on docs/reference/intro.asciidoc:
```diff
@@ -155,81 +184,146 @@ You can use a combination of dynamic and explicit mapping on the same index.
+metadata.
+However, the real power of {es} comes from its advanced search and analytics capabilities.
 
 {es} provides a simple, coherent REST API for managing your cluster and indexing
```
there's that word again

shainaraskas commented on docs/reference/intro.asciidoc:
```diff
@@ -155,81 +184,146 @@ You can use a combination of dynamic and explicit mapping on the same index.
-for your language of choice: Java, JavaScript, Go, .NET, PHP, Perl, Python
-or Ruby.
+and searching your data.
+For testing purposes, you can easily submit requests
```
```suggestion
For testing purposes, you can submit requests
```

shainaraskas commented on docs/reference/intro.asciidoc:
```diff
@@ -155,81 +184,146 @@ You can use a combination of dynamic and explicit mapping on the same index.
+and searching your data.
+For testing purposes, you can easily submit requests
+directly from the command line or through the Dev Tools {kibana-ref}/console-kibana.html[Console] in {kib}.
+From your applications, you can use the
+https://www.elastic.co/guide/en/elasticsearch/client/index.html[{es} clients]
```
```suggestion
From your applications, you can use an
https://www.elastic.co/guide/en/elasticsearch/client/index.html[{es} client]
```

shainaraskas commented on docs/reference/intro.asciidoc:
```diff
@@ -155,81 +184,146 @@ You can use a combination of dynamic and explicit mapping on the same index.
+https://www.elastic.co/guide/en/elasticsearch/client/index.html[{es} clients]
+in your programming language of choice.
+
+<<query-dsl, Query DSL>> is the primary query language for {es} today.
+It's a full-featured JSON-style query language that allows you to perform complex searches and aggregations.
+We'll be using the Query DSL for most of our examples.
```
couple things

1. connect the API + QL.
2. introduce the concept of query languages broadly

3. "we" / "our" is vague here
```suggestion
<<query-dsl, Query DSL>> is the primary query language for {es} today.
It's a full-featured JSON-style query language that allows you to perform complex searches and aggregations.
This documentation uses Query DSL for most query examples.
```

shainaraskas commented on docs/reference/intro.asciidoc:
```diff
@@ -155,81 +184,146 @@ You can use a combination of dynamic and explicit mapping on the same index.
+
+// TODO: link to query languages table overview when we find a good place for it
+
+[TIP]
+====
+<<esql,{esql}>> is our new piped query language (and compute engine) that is initially mainly focused on time-series data like logs and metrics.
+====
```
I'd remove this. if you must keep it, I suggest removing the brackets at all costs, and highlighting why it's not the PRIMARY. it doesn't work on xyz

leemthompo replied:
added "It does not yet support all the features of Query DSL, like full-text search and semantic search."

shainaraskas commented on docs/reference/intro.asciidoc:
```diff
@@ -155,81 +184,146 @@ You can use a combination of dynamic and explicit mapping on the same index.
-matches by the `hire_date` field. Full-text queries find all documents that
-match the query string and return them sorted by _relevance_&mdash;how good a
-match they are for your search terms.
+{es} support a wide range of search techniques including:
```
```suggestion
{es} support a wide range of search techniques, including the following:
```

shainaraskas commented on docs/reference/intro.asciidoc:
```diff
@@ -155,81 +184,146 @@ You can use a combination of dynamic and explicit mapping on the same index.
 
-In addition to searching for individual terms, you can perform phrase searches,
-similarity searches, and prefix searches, and get autocomplete suggestions.
+* <<full-text-queries,*Full-text search*>>. Search text that has been analyzed and indexed to support full-text search based on relevance.
```
using the term in the definition is a little confusing here - consider reword?

shainaraskas commented on docs/reference/intro.asciidoc:
```diff
@@ -155,81 +184,146 @@ You can use a combination of dynamic and explicit mapping on the same index.
+* <<keyword,*Keyword search*>>. Search for exact matches using `keyword` fields.
+* <<semantic-search-semantic-text,*Semantic search*>>. Search `semantic_text` fields using dense or sparse vector search on embeddings generated in your {es} cluster.
+* <<knn-search,*Vector search*>>. Search for similar dense vectors using the kNN algorithm for embeddings generated outside of {es}.
+* <<geo-queries,*Geospatial search*>>. Search for locations and calculate spatial relationships using geospatial queries.
```
this is a better definition you could use in the use cases

shainaraskas commented on docs/reference/intro.asciidoc:
```diff
@@ -155,81 +184,146 @@ You can use a combination of dynamic and explicit mapping on the same index.
-construct <<sql-overview, SQL-style queries>> to search and aggregate data
-natively inside {es}, and JDBC and ODBC drivers enable a broad range of
-third-party applications to interact with {es} via SQL.
+You can also filter data using the Query DSL.
```
to "the" or not to "the"?

shainaraskas commented on docs/reference/intro.asciidoc:
```diff
@@ -155,81 +184,146 @@ You can use a combination of dynamic and explicit mapping on the same index.
-proverbial “needle in a haystack”, aggregations enable you to answer questions
-like:
+{es} enables a host of use cases based on aggregations and analytics.
+These include real-world applications like:
```
links? :) pls

leemthompo replied:
@shainaraskas I'm unsure about this list — feels repetitious versus the main use cases overview. thoughts?

shainaraskas commented on docs/reference/intro.asciidoc:
```diff
@@ -155,81 +184,146 @@ You can use a combination of dynamic and explicit mapping on the same index.
-You can also use aggregations to answer more subtle questions, such as:
+[discrete]
+[[analyze-data-query-dsl]]
+===== Query DSL
```
this heading comes out of nowhere. need to connect query languages with "analyzing your data" - maybe in headings

overall, I think you should make the relationships between the QLs and these sections more consistent.

leemthompo replied:
> overall, I think you should make the relationships between the QLs and these sections more consistent.

💯 

maybe focus on Query DSL and ES|QL, outlining search, filtering, and analysis capabilities for each

and then appendix with table that gives honorable mention to others

shainaraskas reviewed (commented):
looking good! dropped some feedback. will look at the other topics tomorrow hopefully

leemthompo commented on docs/reference/intro.asciidoc:
```diff
@@ -1,68 +1,97 @@
+== {es} basics
+
+This guide covers the essential basics for getting started with {es}.
+If you'd prefer to get started with code examples, jump to our <<quickstart,quickstarts>>.
```
FYI the new "quickstarts" section is all about running code, and not setting up local dev as they are now separate concerns

We chatted about how we want to get a better title for that "quickstarts" section too

modulo that, everything in this PR can be reworded more gooder

leemthompo replied:
I'll take suggestion 1 for the moment

leemthompo replied:
will go with "If you'd prefer to start working with {es} right away, then jump to our <<quickstart,quickstarts>> for hands-on code examples."

shainaraskas replied:
where did the local dev go?

leemthompo replied:
right, there's a prerequisites section in quickstarts overview that links to local dev and deployment options, but could spell it out here too 👍

leemthompo replied:
`If you'd prefer to start working with {es} right away, set up a <<run-elasticsearch-locally,local dev environment>> and jump to our <<quickstart,hands-on code examples>> .`

marciw replied:
I find it a bit odd that throughout the docs, we say "our" examples, "our" tutorials... "we" shouldn't pull focus in things that aren't really about Elastic. 🙂  I'd just say "some" instead of "our"

leemthompo replied:
I'll do my part and remove "our" usages in this PR 🫡

leemthompo commented on docs/reference/intro.asciidoc:
```diff
@@ -1,68 +1,97 @@
-== What is {es}?
+== {es} basics
+
+This guide covers the essential basics for getting started with {es}.
```
💯

leemthompo commented on docs/reference/intro.asciidoc:
```diff
@@ -1,68 +1,97 @@
+* <<documents-indices>>: Understand {es}'s most important primitives and how it stores data.
+* <<es-ingestion-overview>>: Understand your options for ingesting data into {es}.
+* <<search-analyze>>: Understand your options for searching and analyzing data in {es}.
+* <<scalability>>: Learn the essentials about moving your {es} deployment to production.
```
```suggestion
* <<scalability>>: Understand the basic concepts required for moving your {es} deployment to production.
```

leemthompo commented on docs/reference/intro.asciidoc:
```diff
@@ -1,68 +1,97 @@
+* <<documents-indices>>: Understand {es}'s most important primitives and how it stores data.
+* <<es-ingestion-overview>>: Understand your options for ingesting data into {es}.
+* <<search-analyze>>: Understand your options for searching and analyzing data in {es}.
+* <<scalability>>: Learn the essentials about moving your {es} deployment to production.
```
👍 revised wording

leemthompo commented on docs/reference/intro.asciidoc:
```diff
@@ -1,68 +1,97 @@
 
-* *Monitor log and event data*: Store logs, metrics, and event data for observability and security information and event management (SIEM).
-* *Build search applications*: Add search capabilities to apps or websites, or build search engines over internal data.
+**Observability**
```
>  I know you are trying to be light on links here.

would definitely like to have more links here tho eventually just need to loop in the right folks

leemthompo commented on docs/reference/intro.asciidoc:
```diff
@@ -1,68 +1,97 @@
+
+**Quick start option**
+
+* <<run-elasticsearch-locally,*Local dev*>>: Get started quickly with a minimal local Docker setup. Recommended if you like getting started in a local environment.
```
yeah might just remove that second sentence entirely

"run local dev if you like local dev" is a bit tautological haha

leemthompo replied:
rewording to "Get started quickly with a minimal local Docker setup for development and testing."

leemthompo commented on docs/reference/intro.asciidoc:
```diff
@@ -1,68 +1,97 @@
 * {serverless-docs}/general/sign-up-trial[*Elastic Cloud Serverless* (technical preview)]: Create serverless projects for autoscaled and fully managed {es} deployments. Sign up for a https://cloud.elastic.co/serverless-registration[14-day free trial].
 
-**Advanced deployment options**
+**Advanced options**
```
cool yeah agree this list is prob ok for now

leemthompo commented on docs/reference/intro.asciidoc:
```diff
@@ -84,7 +113,7 @@ Some key facts about indices:
 
 * An index is a collection of documents
 * An index has a unique name
```
I like it I'll draft some additional color here

leemthompo replied:
moved to prose but now thinking maybe it is needlessly detailed at this point

leemthompo replied:
I changed my mind again in the blink of an eye

leemthompo commented on docs/reference/intro.asciidoc:
```diff
@@ -129,7 +158,7 @@ A simple {es} document might look like this:
-In {es}, metadata fields are prefixed with an underscore.
+In {es}, <<mapping-fields,metadata fields>> are prefixed with an underscore.
 
 The most important metadata fields are:
```
yeah agree the goal was just to say here are some basics about what an elasticsearch document looks like if you have never seen one before, but _source is not super imp  

> consider dropping the list of most important metadata fields to avoid finger on the scale

💯

leemthompo commented on docs/reference/intro.asciidoc:
```diff
@@ -155,81 +184,146 @@ You can use a combination of dynamic and explicit mapping on the same index.
 This is useful when you have a mix of known and unknown fields in your data.
```
fair enough here I guess it might be "unexpected" for the naive user, so it's more of a statement about a novice user's epistemology — "which you might find unexpected" — but agree this can be tightened. I don't love _imprecise_ either. We also don't need to be exhaustive here.

I'd rather be concrete like "can lead to suboptimal mappings because certain field types might not be inferred correctly".

> re: explicit on like 150 ... it's not just about control, it's also about making your data more functional and opening up additional use cases.

ok but we link to the main mappings section for details, this is just an overview at end of the day, we don't need 100% exhaustive summaries of capabilities 

in my little head "control" is a decent encapsulation of the advantage of explicit mappings in this context but happy to reword

leemthompo replied:
went with 

- "This is great for getting started quickly, but can yield suboptimal results for complex data due to automatic field type inference."
- "Recommended for production use cases, because you have full control over how your data is indexed to suit your specific use case."

leemthompo committed:
Apply suggestions from code review

Co-authored-by: shainaraskas <58563081+shainaraskas@users.noreply.github.com>

leemthompo committed:
Apply suggestions from code review

Co-authored-by: shainaraskas <58563081+shainaraskas@users.noreply.github.com>

leemthompo reviewed (commented):
I did some comment responses as a review by accident, just to make this even messier

leemthompo commented on docs/reference/intro.asciidoc:
```diff
@@ -84,7 +113,7 @@ Some key facts about indices:
 
 * An index is a collection of documents
 * An index has a unique name
-* An index can also be referred to by an alias
+* An index can also be referred to by an <<aliases,alias>>
 * An index has a mapping that defines the schema of its documents
```
I like poetry, but agree :)

leemthompo committed:
Updates per review

leemthompo commented:
thanks @shainaraskas 🙏  I've tried to tackle all of the immediately actionable suggestions.

Big picture, unresolved is

-  Agreeing on our approach to the "Add data" section here versus cloud page 
- Need to refocus the "Search and analyze data" sections with clarity on query languages

leemthompo committed:
Refactor search and analyze section, add back QL table

leemthompo committed:
Stop saying -our- all the time

leemthompo commented on docs/reference/intro.asciidoc:
```diff
@@ -1,68 +1,97 @@
+**Search**
+
+* *Full-text search*: Fast, relevant full-text search using inverted indexes, tokenization, and text analysis.
+* *Vector database*: Store and search vectorized data, and create vector embeddings with built-in and third-party natural language processing (NLP) models.
```
```suggestion
* *Vector database*: Store and search vectorized data, and create vector embeddings with built-in and third-party natural language processing (NLP) models.
* *Semantic search*: Understand the intent and contextual meaning behind search queries using tools like synonyms, dense vector embeddings, and learned sparse query/document expansion.
```

leemthompo committed:
Add semantic search to use cases

suggested by @szabosteve

leemthompo committed:
Merge branch 'main' into 101-section

leemthompo committed:
Merge branch 'main' into 101-section

leemthompo committed:
Delete duplicated include

elasticsearchmachine commented:
Pinging @elastic/es-docs (Team:Docs)

leemthompo commented:
> I would also like more code examples. Add Data and Search pages are going to be crucial ones so would be nice to show how simple it is to ingest and search in two REST calls. Would be nice if the links to full-text, vector search, semantic-search go straight to code examples for example.

I understand and agree in many respects. But for adding data we have to walk the tightrope of first telling people all their options, which we don't currently do in a digestible way. If we were to start adding code examples in that overview it would disrupt the flow. But can definitely link more explicitly in this section to the [First steps with ES quickstart](https://app.markup.io/markup/41a10056-a6b7-4c0b-a5bd-1caeda4276d0?view_mode=Desktop) again, where you have that hands-on primer.

We do have the escape hatch on the basics landing page that says "If you’d prefer to start working with Elasticsearch right away, set up a [local dev environment](https://e11f4504-57a2-41e1-be06-f74b1abf8a63.p.markup.io/guide/en/elasticsearch/reference/master/run-elasticsearch-locally.html "Run Elasticsearch locally in Docker") and jump to [hands-on code examples](https://e11f4504-57a2-41e1-be06-f74b1abf8a63.p.markup.io/guide/en/elasticsearch/reference/master/quickstart.html "Quickstarts")."

In next iteration definitely want all of this quickstarts to start landing in short order, and then update the links in the [Search and analyze](https://app.markup.io/markup/41a10056-a6b7-4c0b-a5bd-1caeda4276d0?view_mode=Desktop) section to jump there, rather than just the abstract docs. There I fully agree, but this needs to be iterative: with the hands-on stuff coming next, and in each of those concrete quickstarts we can link to the relevant notebook and blogs too. But we can't do it _all_ on these overview pages. This PR is already trying to do a _lot_.

Also the scope of this is slightly wider than just the search dev, and hence we need to be a little broader in how we couch some of this information.

joemcelroy commented:
Great to hear and agree with you on the direction with going text heavy first and then doing another layer for code.

leemthompo committed:
Some updates based on Serena feedback

- Added details to the recommendation for using Elastic Observability for timestamped data.
- Clarified the potential suboptimal results of dynamic mapping for specific use cases.
- Enhanced the description of general content ingestion options, including a note for Python developers.

leemthompo committed:
Rename quickstart section, add vector example

leemthompo committed:
Add snippet test teardown, uniformize quick start term

leemthompo committed:
Keep search your data where it is in nav for now

leemthompo committed:
Make it clear esql has new goodness every release

leemthompo committed:
API quick starts is better name

shainaraskas commented on docs/reference/intro.asciidoc:
```diff
@@ -146,90 +170,197 @@ A mapping defines the <<mapping-types,data type>> for each field, how the field
 
-* <<mapping-dynamic, Dynamic mapping>>: Let {es} automatically detect the data types and create the mappings for you. This is great for getting started quickly, but can lead to unexpected results for complex data.
-* <<mapping-explicit, Explicit mapping>>: Define the mappings up front by specifying data types for each field. Recommended for production use cases, because you have much more control over how your data is indexed.
+* <<mapping-dynamic, Dynamic mapping>>: Let {es} automatically detect the data types and create the mappings for you. This is great for getting started quickly, but might yield suboptimal results for your specific use case due to automatic field type inference.
```
:2c: avoid `this is great for` due to vagueness

```suggestion
* <<mapping-dynamic, Dynamic mapping>>: Let {es} automatically detect the data types and create the mappings for you. Dynamic mapping is useful for getting started quickly, but might yield suboptimal results for your specific use case due to automatic field type inference.
```

leemthompo replied:
I would contend that "useful" is equally vague but we're getting into the nitty weeds

Maybe we can go from "Great to good"? Still vague but feels nicer in active voice?

shainaraskas commented on docs/reference/intro.asciidoc:
```diff
@@ -146,90 +170,197 @@ A mapping defines the <<mapping-types,data type>> for each field, how the field
+=== Add data to {es}
+
+There are multiple ways to ingest data into {es}.
+The option that you choose depends on whether you're working with timestamped data, non-timestamped data, where the data is coming from, its complexity, and more.
```
gramma 

```suggestion
The option that you choose depends on whether you're working with timestamped data or non-timestamped data, where the data is coming from, its complexity, and more.
```

shainaraskas commented on docs/reference/intro.asciidoc:
```diff
@@ -146,90 +170,197 @@ A mapping defines the <<mapping-types,data type>> for each field, how the field
+==== General content
+
+General content is data that does not have a timestamp.
+This could be data like vector embeddings, website content, product catalogs, or more.
```
```suggestion
This could be data like vector embeddings, website content, product catalogs, and more.
```

leemthompo replied:
"or more" 😆

shainaraskas commented on docs/reference/intro.asciidoc:
```diff
@@ -146,90 +170,197 @@ A mapping defines the <<mapping-types,data type>> for each field, how the field
+This could be data like vector embeddings, website content, product catalogs, or more.
+For general content, you have the following options for adding data to {es} indices:
+
+* <<docs,API>>: Use the {es} <<docs,Document APIs>> to index documents directly, using the Dev Tools {kibana-ref}/console-kibana.html[Console], cURL.
```
```suggestion
* <<docs,API>>: Use the {es} <<docs,Document APIs>> to index documents directly, using the Dev Tools {kibana-ref}/console-kibana.html[Console], or cURL.
```

shainaraskas commented on docs/reference/intro.asciidoc:
```diff
@@ -146,90 +170,197 @@ A mapping defines the <<mapping-types,data type>> for each field, how the field
+For general content, you have the following options for adding data to {es} indices:
+
+* <<docs,API>>: Use the {es} <<docs,Document APIs>> to index documents directly, using the Dev Tools {kibana-ref}/console-kibana.html[Console], cURL.
+** You can use https://www.elastic.co/guide/en/elasticsearch/client/index.html[{es} programming language clients] to index documents in your programming language of choice. For Python devs, check out the `elasticsearch-labs` repo for various https://github.com/elastic/elasticsearch-labs/tree/main/notebooks/search/python-examples[example notebooks]. 
```
why so much emphasis on python here?

leemthompo replied:
GenAI is a thing, plus notebooks are currently our best e2e how-tos

shainaraskas commented on docs/reference/intro.asciidoc:
```diff
@@ -146,90 +170,197 @@ A mapping defines the <<mapping-types,data type>> for each field, how the field
+For general content, you have the following options for adding data to {es} indices:
+
+* <<docs,API>>: Use the {es} <<docs,Document APIs>> to index documents directly, using the Dev Tools {kibana-ref}/console-kibana.html[Console], cURL.
+** You can use https://www.elastic.co/guide/en/elasticsearch/client/index.html[{es} programming language clients] to index documents in your programming language of choice. For Python devs, check out the `elasticsearch-labs` repo for various https://github.com/elastic/elasticsearch-labs/tree/main/notebooks/search/python-examples[example notebooks]. 
```
consider making the client connection to the API a little clearer. Also, I don't think they're "es programming language clients", but instead clients offered in various languages.

```suggestion
+
If you're building a website or app, then you can call Elasticsearch APIs using an https://www.elastic.co/guide/en/elasticsearch/client/index.html[{es} client] in the programming language of your choice. If you use the Python client, then check out the `elasticsearch-labs` repo for various https://github.com/elastic/elasticsearch-labs/tree/main/notebooks/search/python-examples[example notebooks]. 
```

shainaraskas commented on docs/reference/intro.asciidoc:
```diff
@@ -146,90 +170,197 @@ A mapping defines the <<mapping-types,data type>> for each field, how the field
+
+* <<docs,API>>: Use the {es} <<docs,Document APIs>> to index documents directly, using the Dev Tools {kibana-ref}/console-kibana.html[Console], cURL.
+** You can use https://www.elastic.co/guide/en/elasticsearch/client/index.html[{es} programming language clients] to index documents in your programming language of choice. For Python devs, check out the `elasticsearch-labs` repo for various https://github.com/elastic/elasticsearch-labs/tree/main/notebooks/search/python-examples[example notebooks]. 
+* {kibana-ref}/connect-to-elasticsearch.html#upload-data-kibana[File upload]: Use the {kib} file uploader to upload and index CSV, JSON, and log files.
```
this one has a super clear use case we might point to (exploration / index setup)

leemthompo replied:
vomited up a draft of that

shainaraskas commented on docs/reference/intro.asciidoc:
```diff
@@ -146,90 +170,197 @@ A mapping defines the <<mapping-types,data type>> for each field, how the field
+* <<docs,API>>: Use the {es} <<docs,Document APIs>> to index documents directly, using the Dev Tools {kibana-ref}/console-kibana.html[Console], cURL.
+** You can use https://www.elastic.co/guide/en/elasticsearch/client/index.html[{es} programming language clients] to index documents in your programming language of choice. For Python devs, check out the `elasticsearch-labs` repo for various https://github.com/elastic/elasticsearch-labs/tree/main/notebooks/search/python-examples[example notebooks]. 
+* {kibana-ref}/connect-to-elasticsearch.html#upload-data-kibana[File upload]: Use the {kib} file uploader to upload and index CSV, JSON, and log files.
+* {kibana-ref}/connect-to-elasticsearch.html#_add_sample_data[Sample data]: Load sample data sets into your {es} cluster using {kib}.
```
I wouldn't put this item inline. instead, consider putting it above the general/timestamped buckets (some of this data is timestamped). I'd also frame this as "if you just want to play around, consider using our sample data to start working with es"

leemthompo replied:
oh yeah good point

shainaraskas commented on docs/reference/intro.asciidoc:
```diff
@@ -146,90 +170,197 @@ A mapping defines the <<mapping-types,data type>> for each field, how the field
 
+// New html page
+[[es-ingestion-overview]]
+=== Add data to {es}
```
the changes that you made to this page are so so valuable! I LOVE being able to understand how to get data into ES at a high level. would have been a game changer for March Shaina.

karenzone replied:
The origin of this content has been available for a couple of years, but somewhat hidden: https://www.elastic.co/guide/en/cloud/current/ec-cloud-ingest-data.html 

I've been using this starter content as the foundation of my ingest experience work, which means we'll have some duplication to sort out. We knew that there would be overlaps, so no big surprise there.  We'll figure out the best way to communicate this info to users, and do the right thing.

shainaraskas reviewed (commented):
submitting a little batch of comments - attempting to finish a comprehensive review today

shainaraskas commented on docs/reference/run-elasticsearch-locally.asciidoc:
```diff
@@ -12,21 +12,8 @@ The instructions on this page are for *local development only*. Do not use these
+(If you don't need the {kib} UI, you can skip those instructions.)
 
 [discrete]
 [[local-dev-prerequisites]]
```
docker network should also be marked as optional ... but unfortunately the rest of the commands depend on it.

they might be inextricable in this version of the tutorial

leemthompo replied:
again very soon moot

shainaraskas commented on docs/reference/run-elasticsearch-locally.asciidoc:
```diff
@@ -12,21 +12,8 @@ The instructions on this page are for *local development only*. Do not use these
-* Follow a tutorial or guide that requires an {es} cluster, like our <<getting-started,quick start guide>>.
-* Experiment with the {es} APIs using different tools, like the Dev Tools Console, cURL, or an Elastic programming language client.
-* Quickly spin up an {es} cluster to test an executable https://github.com/elastic/elasticsearch-labs/tree/main/notebooks#readme[Python notebook] locally.
+Use this setup if you want to quickly spin up {es} (and {kib}) in Docker for local development or testing.
+(If you don't need the {kib} UI, you can skip those instructions.)
```
consider doing a tiiiiny sales pitch for kibana here.

leemthompo replied:
this whole page will disappear VERY soon in favor of `start-local` so not worth thinking too deeply about

shainaraskas commented on docs/reference/run-elasticsearch-locally.asciidoc:
```diff
@@ -118,12 +105,12 @@ When you access {kib}, use `elastic` as the username and the password you set ea
 [discrete]
 [[local-dev-connecting-clients]]
-== Connecting to {es} with language clients
+=== Connecting to {es} with language clients
```
this is the next step, so it should use the same tense

this makes me think: should these steps be numbered so people don't lose their spot? 

```suggestion
=== Connect to {es} with language clients
```

shainaraskas commented on docs/reference/run-elasticsearch-locally.asciidoc:
```diff
@@ -118,12 +105,12 @@ When you access {kib}, use `elastic` as the username and the password you set ea
-== Connecting to {es} with language clients
+=== Connecting to {es} with language clients
 
 To connect to the {es} cluster from a language client, you can use basic authentication with the `elastic` username and the password you set in the environment variable.
```
we chatted about this before, but imo you should only hide the examples and not the credentials.

shainaraskas commented on docs/reference/tab-widgets/api-call.asciidoc:
```diff
@@ -1,5 +1,5 @@
 // tag::cloud[]
-**Use {kib}**
+**Option 1: Use {kib}**
```
not sure where this file is being used so I didn't preview it, but nothing looks weird at a glance.

shainaraskas reviewed (commented):
AMAZING!!!!! imo this is nearly done! these are just polish-type feedback that would probably make sense to quickly tackle since these pages are so close to the top of the funnel. 

thanks for your patience with my 🐌 review!

karenzone reviewed (commented):
@leemthompo and everyone else who has contributed to this effort: Amazing work! So exciting to see this. 

As noted inline and as expected, we have some overlap between the ingest portion of this project and the ingest experience docs that I'm working on. I look forward to working with you on getting the right info to the user at the right time to deliver the best information experience we can.

leemthompo commented on docs/reference/index.asciidoc:
```diff
@@ -6,10 +6,10 @@ include::links.asciidoc[]
 
-include::intro.asciidoc[]
-
 include::release-notes/highlights.asciidoc[]
 
+include::intro.asciidoc[]
+
 include::quickstart/index.asciidoc[]
```
1. intro.asciidoc (basics) contains "run elasticsearch" options  and having running instance is prerequisite for quickstarts
    - there's going to be a lot more "quick starts" here so we need to decouple installation and these guides
3. this is asciidoc weirdness wanted to move it beside release notes but was too much fussiness to add to this pile of fussiness

leemthompo replied:
wanted basics and quick starts to be beside each other rather than have what's new in between as it is today

leemthompo commented on docs/reference/intro.asciidoc:
```diff
@@ -1,68 +1,98 @@
+**Observability**
+
+* *Logs, metrics, and traces*: Collect, store, and analyze logs, metrics, and traces from applications, systems, and services.
+* *Application performance monitoring (APM)*: Monitor and analyze application performance data.
```
I'll take "business critical" with one eyebrow very slightly raised

leemthompo commented on docs/reference/intro.asciidoc:
```diff
@@ -1,68 +1,98 @@
+
+* *Full-text search*: Fast, relevant full-text search using inverted indexes, tokenization, and text analysis.
+* *Vector database*: Store and search vectorized data, and create vector embeddings with built-in and third-party natural language processing (NLP) models.
+* *Semantic search*: Understand the intent and contextual meaning behind search queries using tools like synonyms, dense vector embeddings, and learned sparse query/document expansion.
```
I'm editing your suggestion to `query-document` pair, which is the form I was looking for, [reference](https://www.researchgate.net/figure/Examples-of-Query-Document-Pairs_tbl4_381318617).

leemthompo commented on docs/reference/intro.asciidoc:
```diff
@@ -1,68 +1,98 @@
+* *Full-text search*: Fast, relevant full-text search using inverted indexes, tokenization, and text analysis.
+* *Vector database*: Store and search vectorized data, and create vector embeddings with built-in and third-party natural language processing (NLP) models.
+* *Semantic search*: Understand the intent and contextual meaning behind search queries using tools like synonyms, dense vector embeddings, and learned sparse query/document expansion.
+* *Hybrid search*: Combine full-text search with vector search using state-of-the-art ranking algorithms.
```
we have to color outside the lines of consistency a tiny bit, just because vectorDB is a keyword

leemthompo commented on docs/reference/intro.asciidoc:
```diff
@@ -1,68 +1,98 @@
+* *Vector database*: Store and search vectorized data, and create vector embeddings with built-in and third-party natural language processing (NLP) models.
+* *Semantic search*: Understand the intent and contextual meaning behind search queries using tools like synonyms, dense vector embeddings, and learned sparse query/document expansion.
+* *Hybrid search*: Combine full-text search with vector search using state-of-the-art ranking algorithms.
+* *Search applications*: Add hybrid search capabilities to apps or websites, or build enterprise search engines over your organization's internal data sources.
```
```suggestion
* *Build search experiences*: Add hybrid search capabilities to apps or websites, or build enterprise search engines over your organization's internal data sources.
```

leemthompo commented on docs/reference/intro.asciidoc:
```diff
@@ -1,68 +1,98 @@
+* *Vector database*: Store and search vectorized data, and create vector embeddings with built-in and third-party natural language processing (NLP) models.
+* *Semantic search*: Understand the intent and contextual meaning behind search queries using tools like synonyms, dense vector embeddings, and learned sparse query/document expansion.
+* *Hybrid search*: Combine full-text search with vector search using state-of-the-art ranking algorithms.
+* *Search applications*: Add hybrid search capabilities to apps or websites, or build enterprise search engines over your organization's internal data sources.
```
definitely mean the generic thang, updating to "Build search experiences" for maximum augmented disambiguation (MAD)

leemthompo commented on docs/reference/intro.asciidoc:
```diff
@@ -1,68 +1,98 @@
+* *Semantic search*: Understand the intent and contextual meaning behind search queries using tools like synonyms, dense vector embeddings, and learned sparse query/document expansion.
+* *Hybrid search*: Combine full-text search with vector search using state-of-the-art ranking algorithms.
+* *Search applications*: Add hybrid search capabilities to apps or websites, or build enterprise search engines over your organization's internal data sources.
+* *Retrieval augmented generation (RAG)*: Use {es} as a retrieval engine to update and augment Generative AI models.
```
agree

leemthompo replied:
(me talking to myself here to improve this line) RAG is more than just about returning better results (R), it's also doing stuff with those results (G)

leemthompo commented on docs/reference/intro.asciidoc:
```diff
@@ -1,68 +1,98 @@
+* *Semantic search*: Understand the intent and contextual meaning behind search queries using tools like synonyms, dense vector embeddings, and learned sparse query/document expansion.
+* *Hybrid search*: Combine full-text search with vector search using state-of-the-art ranking algorithms.
+* *Search applications*: Add hybrid search capabilities to apps or websites, or build enterprise search engines over your organization's internal data sources.
+* *Retrieval augmented generation (RAG)*: Use {es} as a retrieval engine to update and augment Generative AI models.
```
```suggestion
* *Retrieval augmented generation (RAG)*: Use {es} as a retrieval engine to supplement generative AI models with more relevant, up-to-date, or proprietary data for a range of use cases.
```

leemthompo commented on docs/reference/intro.asciidoc:
```diff
@@ -129,11 +155,9 @@ A simple {es} document might look like this:
 ==== Data and metadata
 
 An indexed document contains data and metadata.
```
I refactored your first sentence and not sure if your second sentence adds a lot of value in an intro.

leemthompo replied:
could maybe do a nicer job of annotating a document in hands-on API companion to provide a concrete example of what you were alluding to in "Some metadata fields dictate how the document is processed or stored."

leemthompo commented on docs/reference/intro.asciidoc:
```diff
@@ -146,90 +170,197 @@ A mapping defines the <<mapping-types,data type>> for each field, how the field
+
+* <<docs,API>>: Use the {es} <<docs,Document APIs>> to index documents directly, using the Dev Tools {kibana-ref}/console-kibana.html[Console], cURL.
+** You can use https://www.elastic.co/guide/en/elasticsearch/client/index.html[{es} programming language clients] to index documents in your programming language of choice. For Python devs, check out the `elasticsearch-labs` repo for various https://github.com/elastic/elasticsearch-labs/tree/main/notebooks/search/python-examples[example notebooks]. 
+* {kibana-ref}/connect-to-elasticsearch.html#upload-data-kibana[File upload]: Use the {kib} file uploader to upload and index CSV, JSON, and log files.
```
```suggestion
* {kibana-ref}/connect-to-elasticsearch.html#upload-data-kibana[File upload]: Use the {kib} file uploader to upload and index various file formats, including CSV, JSON, and log files. The GUI makes it easy to add data and guides you through the index setup process. 
```

leemthompo commented on docs/reference/intro.asciidoc:
```diff
@@ -146,90 +170,197 @@ A mapping defines the <<mapping-types,data type>> for each field, how the field
 // New html page
 [[search-analyze]]
-=== Search and analyze
+=== Search and analyze data
```
I'm hesitant to start refactoring this again because this PR is out of control

I totally agree with the flow, but honestly I'm not sure how well any of this flows. 

> I think this page is better

This is the acceptance criteria

leemthompo replied:
All of this stuff will probably be revisited one way or another quite soon so I don't think we need to polish this excessively immediately

shainaraskas replied:
lgtm

leemthompo commented on docs/reference/intro.asciidoc:
```diff
@@ -146,90 +170,197 @@ A mapping defines the <<mapping-types,data type>> for each field, how the field
+
+
+| <<eql,EQL>>
+| Event Query Language (EQL) is a query language for event-based time series data. Data must contain an `@timestamp` field to use EQL.
```
I found "the" workaround

leemthompo commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -1,47 +1,20 @@
-== Quick start guide
+== Quick start: Add data using API
+++++
+<titleabbrev>Basics: Add data using API</titleabbrev>
```
🫡

leemthompo commented on docs/reference/intro.asciidoc:
```diff
@@ -146,90 +170,197 @@ A mapping defines the <<mapping-types,data type>> for each field, how the field
+| Name | Description | Use cases | API endpoint
+
+| <<query-dsl,Query DSL>>
+| Primary query language for {es}. Powerful and flexible JSON-style language that enables complex queries.
```
Unfortunately I read [a book about the Beaufort Scale](https://huler.weebly.com/defining-the-wind.html) and the concision of the language in that original scale has colored my entire approach to grammatical completeness in tables 😃 


[Snippet from scale](https://github.com/user-attachments/assets/8b5c91e6-df52-4f15-a8fa-194ce8096e60)

leemthompo commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -1,47 +1,20 @@
+++++
 
-This guide helps you learn how to:
+In this quickstart guide you'll learn how to:
```
the last instance of "quickstart" standing

leemthompo commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -212,6 +192,42 @@ You should receive a response indicating there were no errors.
+[[add-vector-content]]
+==== Add vector content
+
+If you have your own dense vector embeddings, you can index them along with your documents. This example shows how to add a document with an embedding to the `books` index.
```
agree, it's a tip of the cap to the importance of ability to add vectors to elasticsearch 

you can ingest a vector just as easily as you can add anything else

but then for symmetry I had to introduce querying that dense vector and here I agree ventured far deeper than the first quick start should

to be clear on scope of quick starts though, they should be code companions to prose, ideally we do zero conceptual explanation in those, but they're tied at the hip to their prose sibling

leemthompo replied:
tl;dr I'll remove vectors

leemthompo replied:
getting ahead of my skis because next step is a set of quick starts

leemthompo replied:
> I question that. if they don't do ANY conceptual explanation, they need to be sitting with their conceptual counterparts, not extracted into a dedicated quickstart section. even a sentence or two about what's being done in the code would be enough imo

haha yeah you're totally right I have a draft here and there's plenty of prose glue 😃

leemthompo commented on docs/reference/run-elasticsearch-locally.asciidoc:
```diff
@@ -1,7 +1,7 @@
+== Run {es} locally in Docker
 ++++
-<titleabbrev>Local dev setup (Docker)</titleabbrev>
+<titleabbrev>Start in local dev</titleabbrev>
```
```suggestion
<titleabbrev>Run {es} locally</titleabbrev>
```

leemthompo commented on docs/reference/run-elasticsearch-locally.asciidoc:
```diff
@@ -1,7 +1,7 @@
+== Run {es} locally in Docker
 ++++
-<titleabbrev>Local dev setup (Docker)</titleabbrev>
+<titleabbrev>Start in local dev</titleabbrev>
```
🫡

leemthompo committed:
Apply suggestions from code review

Co-authored-by: shainaraskas <58563081+shainaraskas@users.noreply.github.com>

leemthompo commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -254,31 +270,23 @@ GET books/_search
+==== Vector search
 
-* <<search-with-elasticsearch>>. Jump here to learn about exact value search, full-text search, vector search, and more, using the <<search-search,search API>>.
+You can use the <<knn-retriever,`knn` retriever>> to perform a k-nearest neighbor search on the `embedding` field.
```
removing vectors here

leemthompo committed:
Move sample data to tip

leemthompo committed:
Remove vectors from first quick start, update local dev verbiage

leemthompo committed:
Add short definition of timestamped data

leemthompo committed:
Mention quick starts use queryDSL unless otherwise noted

leemthompo commented on docs/reference/quickstart/index.asciidoc:
```diff
@@ -1,10 +1,29 @@
+
+[discrete]
+[[quickstart-python-links]]
+== Working in Python
```
@shainaraskas once have healthy list of API quick starts here can shrink Python callout to one line. It's just for the time being we have (checks catalog) "1" quick start :)

leemthompo committed:
Fix url

leemthompo committed:
fix typo

leemthompo committed:
Update query languages section

leemthompo committed:
An -> a

leemthompo committed:
Tweaky McTweakface

shainaraskas commented on docs/reference/run-elasticsearch-locally.asciidoc:
```diff
@@ -12,21 +12,8 @@ The instructions on this page are for *local development only*. Do not use these
 While this approach is convenient for experimenting and learning, you should never run the service in this way in a production environment.
```
hmm one of us edited this to make it grammatically incorrect - "the Elasticsearch"

shainaraskas commented on docs/reference/intro.asciidoc:
```diff
@@ -146,90 +170,216 @@ A mapping defines the <<mapping-types,data type>> for each field, how the field
+* <<docs,API>>: Use the {es} <<docs,Document APIs>> to index documents directly, using the Dev Tools {kibana-ref}/console-kibana.html[Console], or cURL.
++
+If you're building a website or app, then you can call Elasticsearch APIs using an https://www.elastic.co/guide/en/elasticsearch/client/index.html[{es} client] in the programming language of your choice. If you use the Python client, then check out the `elasticsearch-labs` repo for various https://github.com/elastic/elasticsearch-labs/tree/main/notebooks/search/python-examples[example notebooks]. ...
+* {kibana-ref}/connect-to-elasticsearch.html#upload-data-kibana[File upload]: Use the {kib} file uploader to upload and index various file formats, including CSV, JSON, and log files. The GUI guides you through the index setup process, making it easier to prepare your data for search and exploration."
```
perhaps also useful for uploading single files, rather than ongoing processing?

leemthompo replied:
done

shainaraskas commented on docs/reference/intro.asciidoc:
```diff
@@ -146,90 +170,216 @@ A mapping defines the <<mapping-types,data type>> for each field, how the field
+directly executed within {es} itself.
+{esql} syntax can also be used within various {kib} tools.
+
+The <<esql-rest,`_query` endpoint>> accepts queries written in {esql} syntax.
```
🎉

leemthompo committed:
Clarify file uploader use case

leemthompo committed:
Delegate file format info to uploader docs, pdf support is coming soon for example

leemthompo committed:
Local dev-elopment

shainaraskas commented on docs/reference/intro.asciidoc:
```diff
@@ -184,34 +184,42 @@ This is useful when you have a mix of known and unknown fields in your data.
 [[es-ingestion-overview-timestamped]]
 ==== Timestamped data
 
+Timestamped data in {es} refers to datasets that include a timestamp field, typically named `@timestamp` when using the https://www.elastic.co/guide/en/ecs/current/ecs-reference.html[Elastic Common Schema (ECS)].
```
```suggestion
Timestamped data in {es} refers to datasets that include a timestamp field. If you use the https://www.elastic.co/guide/en/ecs/current/ecs-reference.html[Elastic Common Schema (ECS)],  this file is named `@timestamp`.
```

shainaraskas commented on docs/reference/intro.asciidoc:
```diff
@@ -354,6 +369,11 @@ Does not yet support full-text search.
 | <<sql-apis,`_sql`>>
 
+| {kibana-ref}/kuery-query.html[Kibana Query Language (KQL)]
+| Kibana Query Language (KQL) is a text-based query language for filtering data that is only available in the {kib} UI.
```
`only` is probably inaccurate here because it's available through traditional es channels as well, right? 

bad edit: 

```suggestion
| Kibana Query Language (KQL) is a text-based query language for filtering data when you access it through the {kib} UI.
```

shainaraskas commented on docs/reference/quickstart/index.asciidoc:
```diff
@@ -1,28 +1,29 @@
+== Hands-on quick starts
 
-* <<getting-started,First steps with {es}>>. Learn how to add data to {es} and perform basic searches.
+* <<getting-started,Basics: Add data using API>>. Learn how to add data to {es} and perform basic searches.
```
"using API" still feels a little funny to me

shainaraskas replied:
to match sidebar

```suggestion
* <<getting-started,Basics: Add data using APIs>>. Learn how to add data to {es} and perform basic searches.
```

shainaraskas commented on docs/reference/quickstart/index.asciidoc:
```diff
@@ -1,28 +1,29 @@
-* https://github.com/elastic/elasticsearch-labs[`elasticsearch-labs` repository]. Contains a range of Python https://github.com/elastic/elasticsearch-labs/tree/main/notebooks[notebooks] and https://github.com/elastic/elasticsearch-labs/tree/main/example-apps[example apps].
-* https://www.elastic.co/search-labs/tutorials/search-tutorial/welcome[Tutorial]. This walks you through building a complete search solution with {es}, from the ground up using Flask.
+* https://github.com/elastic/elasticsearch-labs[`elasticsearch-labs` repository]: Contains a range of Python https://github.com/elastic/elasticsearch-labs/tree/main/notebooks[notebooks] and https://github.com/elastic/elasticsearch-labs/tree/main/example-apps[example apps].
+* https://www.elastic.co/search-labs/tutorials/search-tutorial/welcome[Tutorial]: This walks you through building a complete search solution with {es}, from the ground up using Flask.
```
```suggestion
* https://www.elastic.co/search-labs/tutorials/search-tutorial/welcome[Tutorial]: This walks you through building a complete search solution with {es} from the ground up using Flask.
```

shainaraskas commented on docs/reference/run-elasticsearch-locally.asciidoc:
```diff
@@ -1,19 +1,21 @@
 
 The instructions on this page are for *local development only*. Do not use these instructions for production deployments, because they are not secure.
-While this approach is convenient for experimenting and learning, you should never run the service in this way in a production environment.
+While this approach is convenient for experimenting and learning, you should never run the Elasticsearch in this way in a production environment.
```
```suggestion
While this approach is convenient for experimenting and learning, you should never run Elasticsearch in this way in a production environment.
```

shainaraskas reviewed (approved):
eyyyyy!

leemthompo committed:
Apply suggestions

Co-authored-by: shainaraskas <58563081+shainaraskas@users.noreply.github.com>

leemthompo commented on docs/reference/quickstart/index.asciidoc:
```diff
@@ -1,10 +1,29 @@
 [[quickstart]]
-= Quickstart
+= API quick starts
```
```suggestion
= Quick starts
```

leemthompo committed:
Future proof quick start section title
